### PR TITLE
Carousel: Enable circular card focus and animation on focus index change

### PR DESCRIPTION
**Pull request by Mitch-At-Work**
## Previous Behavior
Cards could only be navigated to the first/last card then stopped, despite visual order.
Card focus set animation to jump (to prevent auto-scroll issues in browser)

## New Behavior
Card focus can now be navigated circular (to match circular prop)
Card focus will now animate, we may add a reduced motion option in the future which will set jump to true if desired.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.148 MB`<br />`287.39 kB` | `1.148 MB`<br />`287.428 kB` | `133 B` <br />`38 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.14 kB`<br />`20.137 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `218.842 kB`<br />`63.382 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.447 kB`<br />`14.59 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `107.387 kB`<br />`35.758 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/348f67646498dbff929d84d56a5feaa683b0708c'>348f67646498dbff929d84d56a5feaa683b0708c</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-mount_base.html">79</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-mount.html">85</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Avatar-mount_base.html">616</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Avatar-mount.html">611</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Button-mount_base.html">322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Button-mount.html">308</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Field-mount_base.html">1158</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Field-mount.html">1109</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProvider-mount_base.html">777</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProvider-mount.html">697</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-mount_base.html">79</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-mount.html">85</a></td>
            <td>10</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">32</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">33</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">80</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/MakeStyles-mount_base.html">846</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/MakeStyles-mount.html">871</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Persona-mount_base.html">1765</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/Persona-mount.html">1712</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/SpinButton-mount_base.html">1442</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/SpinButton-mount.html">1353</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/SwatchPicker-mount_base.html">1638</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/33015/perf-test-react-components/SwatchPicker-mount.html">1654</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: change/@fluentui-react-carousel-a4f2fe14-2f61-4182-81d9-df0672b90ed5.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-carousel-a4f2fe14-2f61-4182-81d9-df0672b90ed5.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://aka.ms/visual-changes-review?organization=uifabric&project=fabricpublic&prId=33015&commitId=f12ff3634695ca37061677c8d76127edea8c1dbc&env=prod&clientType=FLUENTUI)


  <details><summary><b>Avatar Converged</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Avatar Converged.Badge Mask RTL.chromium.png|4| Changed|

</details>


  <details><summary><b>Drawer</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Drawer.Full Overlay RTL.chromium.png|1166| Changed|
|Drawer.overlay drawer full.chromium.png|2859| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Review by mltejera — APPROVED** [review]

File: packages/react-components/react-carousel/library/src/components/CarouselCard/useCarouselCard.ts [hunk 2]
```diff
@@ -68,10 +69,12 @@ export const useCarouselCard_unstable = (
   const handleFocusCapture = React.useCallback(
     (e: React.FocusEvent) => {
       if (!e.defaultPrevented && isHTMLElement(e.currentTarget) && !isMouseEvent.current) {
-        selectPageByElement(e, e.currentTarget, true);
+        // We want to prevent any browser scroll intervention for 'offscreen' focus
+        containerRef?.current?.scrollTo(0, 0);
```
Comment (layershifter) [inline packages/react-components/react-carousel/library/src/components/CarouselCard/useCarouselCard.ts:73]: Please check if this works with screen readers i.e. VoiceOver, I had bad luck in a similar case recently.

**Pull request merged**